src/id.rs

```diff
@@ -6,64 +6,67 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use crate::parsec;
-use crate::xor_name::XorName;
-use safe_crypto;
-use safe_crypto::{
-    gen_encrypt_keypair, gen_sign_keypair, PublicEncryptKey, PublicSignKey, SecretEncryptKey,
-    SecretSignKey, Signature,
+use crate::{
+    crypto::{encryption, signing},
+    parsec,
+    utils::RngCompat,
+    xor_name::XorName,
 };
+use maidsafe_utilities::serialisation::{deserialise, serialise};
 use serde::de::Deserialize;
 use serde::{Deserializer, Serialize, Serializer};
 use std::fmt::{self, Debug, Display, Formatter};
-use std::ops::RangeInclusive;
+use std::{ops::RangeInclusive, rc::Rc};
 
 /// Network identity component containing name, and public and private keys.
-// FIXME Remove the Clone-ability
 #[derive(Clone)]
```

Comment:
Unfortunately no - we need it because `FullId` is passed to Parsec. I tried to make it safer by putting the secret keys into `Rc`, so no copies are actually made in memory.

Reply:
Ah yes, we will need to probably address that later on but OK for now. I would hope that is not the case fro too much longer mind you. Let's see, but for here we are good for now.